src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -204,7 +204,12 @@ private async Task StartHostAsync(CancellationToken cancellationToken, int attem
 
                 var nextStartupAttemptMode = JobHostStartupMode.Normal;
 
-                if (exc is HostConfigurationException)
+                if (exc is InvalidHostServicesException)
+                {
+                    logger.LogError("Host is being stopped due to invalid service registration. A new host will not be started.");
+                    return;
```

Comment:
For initial release will change this to emit a warning rather than shut down the host so we can gauge the impact.